Source/Core/Common/Debug/Watches.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <cstddef>
+#include <string>
+#include <vector>
+
+#include "Common/CommonTypes.h"
+
+struct Watch
+{
+  u32 address;
+  std::string name;
+  bool is_enabled;
+
+  Watch(u32 address, const std::string& name, bool is_enabled);
+};
+
+class Watches
+{
+public:
+  std::size_t SetWatch(u32 address, const std::string& name);
+  const Watch& GetWatch(std::size_t index) const;
+  const std::vector<Watch>& GetWatches() const;
+  void UnsetWatch(u32 address);
+  std::size_t UpdateWatch(std::size_t index, u32 address, const std::string& name);
```

Comment:
Why return the indexes when it's already provided as part of the arguments? Ditto for the two functions below.

Reply:
It was too be consistent with the ```SetWatch``` method and the only place I used this return value was inside the ```SetWatch``` method. I don't think it would be that useful elsewhere neither, so, I changed it back to void.